### ISSUE TITLE
[BUG] around 200 layer unit tests fail in my hugectr container, pls lend a hand

### ISSUE THREAD

**Issue by heroes999** [NVIDIA-Merlin/HugeCTR#375]
**Describe the bug**
I'm trying to explore hugectr functionalities using hugectr container. I compiled hugectr against my RTX 3090 and run the unit test to verify, but got some 200 layer failure

**To Reproduce**
1. docker run --gpus all --rm -it -p 8888:8888 -p 8797:8787 -p 8796:8786 --ipc=host --cap-add SYS_NICE nvcr.io/nvidia/merlin/merlin-hugectr:22.10 /bin/bash  <-  HugeCTR v4.2 I assume
2. cd /hugectr/build && cmake -DSM=80 .. && make -j16
3. cd bin && ./layers_test

**Expected behavior**
all pass

**Screenshots**
![image](https://user-images.githubusercontent.com/13531432/216276472-cebc135a-50b1-4e05-8f4b-4d41af8efd5f.png)

**Environment (please complete the following information):**
 - OS: Ubuntu 18.04
 - Graphic card: a single NVIDIA GeForce RTX 3090 (is V100/A100 required to enable the most basic functionality of hugectr?)
 - CUDA version: CUDA 11.6
 - Docker image: hugectr 22.10

**Additional context**

**Comment by WonderingWJ** [comment]
@heroes999  Hello! Thank you for your report. I also run the test on 3090 with container `nvcr.io/nvidia/merlin/merlin-hugectr:22.10`. But I can not reproduce the error
```
[ RUN      ] cast_layer.fp32_fp16_32x64
[HCTR][04:47:06.424][DEBUG][RK0][main]: [device 0] allocating 0.0000 GB, available 22.9816
[       OK ] cast_layer.fp32_fp16_32x64 (18 ms)
[ RUN      ] cast_layer.fp32_fp16_64x128
[HCTR][04:47:06.442][DEBUG][RK0][main]: [device 0] allocating 0.0000 GB, available 22.9816
[       OK ] cast_layer.fp32_fp16_64x128 (18 ms)
[ RUN      ] cast_layer.fp32_fp16_128x256
[HCTR][04:47:06.460][DEBUG][RK0][main]: [device 0] allocating 0.0002 GB, available 22.9816
[       OK ] cast_layer.fp32_fp16_128x256 (19 ms)
[ RUN      ] cast_layer.fp32_fp16_256x512
[HCTR][04:47:06.479][DEBUG][RK0][main]: [device 0] allocating 0.0007 GB, available 22.9816
[       OK ] cast_layer.fp32_fp16_256x512 (21 ms)
[ RUN      ] cast_layer.fp16_fp32_32x64
[HCTR][04:47:06.500][DEBUG][RK0][main]: [device 0] allocating 0.0000 GB, available 22.9816
[       OK ] cast_layer.fp16_fp32_32x64 (18 ms)
[ RUN      ] cast_layer.fp16_fp32_64x128
[HCTR][04:47:06.518][DEBUG][RK0][main]: [device 0] allocating 0.0000 GB, available 22.9816
[       OK ] cast_layer.fp16_fp32_64x128 (18 ms)
[ RUN      ] cast_layer.fp16_fp32_128x256
[HCTR][04:47:06.535][DEBUG][RK0][main]: [device 0] allocating 0.0002 GB, available 22.9816
[       OK ] cast_layer.fp16_fp32_128x256 (18 ms)
[ RUN      ] cast_layer.fp16_fp32_256x512
[HCTR][04:47:06.554][DEBUG][RK0][main]: [device 0] allocating 0.0007 GB, available 22.9816
[       OK ] cast_layer.fp16_fp32_256x512 (20 ms)
[----------] 8 tests from cast_layer (150 ms total)
```
Can you provide more info about the error?

**Comment by heroes999** [comment]
It might be my environment problems. The case can pass now. Thanks @WonderingWJ
